src/sparseml/pytorch/sparsification/pruning/modifier_pruning_acdc.py

```diff
@@ -273,3 +278,21 @@ def _finish_on_compression(
                 end_epoch -= 1
 
         return end_epoch
+
+    @staticmethod
+    def _assert_is_integer(x):
+        """
+        Check if x, which is expected to be either a float or int,
+        can be evaluated as int.
+        If True, return and integer, else, raise ValueError.
+
+        :param x: an integer or a float.
+        :return: an integer
+        """
+        if isinstance(x, float) and not x.is_integer():
+            raise ValueError(
+                f"The ACDCPruningModifier assumes that attributes "
+                f"`start_epoch`, `end epoch` and `update frequency` "
+                f"are integers. However, type({x}) is float."
```

Comment:
Might help to reword this error message a bit. As it stands, it suggest to the user that the issue is the `type()` of x and not the value